src/sentry/api/endpoints/project_rule_actions.py

```diff
@@ -60,7 +65,66 @@ def post(self, request: Request, project) -> Response:
             project, platform=project.platform, default="javascript", tagged=True
         )
 
+        if features.has(
+            "projects:verbose-test-alert-reporting", project=project, user=request.user
+        ):
+            return self.execute_future_on_test_event(test_event, rule)
+        else:
+            # Old existing behavior to wrap this in a handler which buries exceptions
+            for callback, futures in activate_downstream_actions(rule, test_event).values():
+                safe_execute(callback, test_event, futures)
+            return Response()
+
+    def execute_future_on_test_event(
+        self,
+        test_event: Event | None,
+        rule: Rule,
+    ) -> Response:
+        """
+        A slightly modified version of utils.safe.safe_execute that handles
+        IntegrationFormErrors, and returns a body with `{ actions: [<error info>] }`.
+
+        This is used in our Alert Rule UI to display errors to the user.
+        """
+        action_exceptions = []
         for callback, futures in activate_downstream_actions(rule, test_event).values():
-            safe_execute(callback, test_event, futures)
+            try:
+                callback(test_event, futures)
+            except Exception as exc:
+                if hasattr(callback, "im_class"):
+                    cls = callback.im_class
```

Comment:
is this in practice needed?  I thought this was removed in python 3 I believe.

Reply:
Oh I'm totally in favor of simplifying this if possible. So is it acceptable to just use `__class__.__name__` now?

Reply:
Yes, I feel.

Reply:
Yeah, you can remove the `im_class` bits that isn't a thing in py3. `callback.__class__` is what you want.